### ISSUE TITLE
`wmllint` doesn't recognize `id=` key in `[set_achievement]` tag

### ISSUE THREAD

**Issue by cooljeanius** [wesnoth/wesnoth#7888]
### Game and System Information

- **Version:** 1.17.20
- **Downloaded from:** Steam
- **OS:** macOS Big Sur

### Description of the bug

MadMax has recently gotten in touch with me about an updated version of "Flight to Freedom" for 1.17 that has achievements. It has code in it like this:
```
                   [set_achievement]
                        content_for=Flight_to_Freedom
                        id=ftf_tribe_mother
                    [/set_achievement]
```
Unfortunately, running `wmllint` on the files with the new achievements like this produces warnings like this:
```
"/Users/ericgallager/Library/Containers/org.wesnoth.Wesnoth/Data/Library/Application Support/Wesnoth_1.16/data/add-ons/Flight_Freedom/scenarios/02a_Rebellion.cfg", line 1240: unknown 'ftf_tribe_mother' referred to by id
```
This seems wrong. Similar to issue #6395, but I think it's different enough to deserve a separate issue.

### Steps to reproduce the behavior

1. Have some WML that uses the `[set_achievement]` key in it
2. Run `wmllint` on it
3. Note warnings

### Expected behavior

`wmllint` should not warn.

**Comment by CelticMinstrel** [comment]
Would be nice if wmllint actually warns when you reference a non-existent achievement… although that would break down if you referenced an achievement from a different add-on…

**Comment by cooljeanius** [comment]
> Would be nice if wmllint actually warns when you reference a non-existent achievement… although that would break down if you referenced an achievement from a different add-on…

...is referencing an achievement from a different add-on intended to be supported behavior?

**Comment by CelticMinstrel** [comment]
Yes, I believe so. That's why achievements must be sorted into groups.

I could maybe see an argument for that being read-only, not sure that's even enforceable though.
